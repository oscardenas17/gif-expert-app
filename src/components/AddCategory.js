import React, { useState } from 'react'

export const AddCategory = () => {

    const [inputValue, setinputValue] = useState('hola mundo');

    const handleInputChange = (e) =>{
        // console.log(e.target.value);
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
    }

    return (
        <form onSubmit = {handleSubmit}>
           <input
                 type="text"
                 value = {inputValue}
                 onChange = {handleInputChange}

           />
        </form>
    )
}
