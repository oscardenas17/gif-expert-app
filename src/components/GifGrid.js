import React from 'react'

export const GifGrid = ( {category}) => {
  
  const getGifs = async() =>{

    const url = 'https://api.giphy.com/v1/gifs/search?q=luffy&limit=10&api_key=NsfM7mcDnuC60HeX26qU7ujruT4Jk2Gg'
    const resp = await fetch(url);
    const {data} = await resp.json();

    const gifs = data.map( img =>{
        return{
            id: img.id,
            title: img.title,
            url: img.images?.downsized_medium.url
        }
    })

    console.log(gifs);


  }
  
  getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
