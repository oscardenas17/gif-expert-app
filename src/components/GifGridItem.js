import React from 'react'

export const GifGridItem = ( {id,title, url} ) => {
    // console.log( id, title, url);

    return (
        <div>
          <img src= {url} alt={title} className="source" />
          <p>{title}</p>
        </div>
    )
}
