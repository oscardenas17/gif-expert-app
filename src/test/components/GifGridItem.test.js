import React from 'react';
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem />', () => {
    
    const title = 'Un titulo';
    const url = 'https//locahost/imagen.jpg';


    test('Debe mostar el componente correctamente', () => {
        
        const wrapper = shallow (<GifGridItem  title={title} url = {url}/>)
        expect(wrapper).toMatchSnapshot();

    })
    

})
