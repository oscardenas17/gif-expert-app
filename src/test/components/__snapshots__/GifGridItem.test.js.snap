// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostar el componente correctamente 1`] = `
<div
  className="card animate__animated animate_fadeIn animate__delay-3s"
>
  <img
    alt="Un titulo"
    src="https//locahost/imagen.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
